backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java

```diff
@@ -50,65 +62,127 @@ private void updatePersonFacility(Person person, UUID facilityId) {
     person.setFacility(facility);
   }
 
-  @AuthorizationConfiguration.RequirePermissionSearchPatients
-  public List<Person> getPatients(UUID facilityId, int pageOffset, int pageSize) {
-    return _repo
-        .findByFacilityAndOrganization(
-            _os.getFacilityInCurrentOrg(facilityId),
-            _os.getCurrentOrganization(),
-            false,
-            PageRequest.of(pageOffset, pageSize, NAME_SORT))
-        .toList();
+  // Specifications filters for queries
+  private Specification<Person> inCurrentOrganizationFilter() {
+    return (root, query, cb) ->
+        cb.equal(root.get(SpecField.ORGANIZATION), _os.getCurrentOrganization());
   }
 
-  @AuthorizationConfiguration.RequirePermissionSearchPatients
-  public long getPatientsCount(UUID facilityId) {
-    return _repo.countAllByFacilityAndOrganization(
-        _os.getFacilityInCurrentOrg(facilityId), _os.getCurrentOrganization(), false);
+  // Note: Patients with NULL facilityIds appear in ALL facilities.
+  private Specification<Person> inFacilityFilter(UUID facilityId) {
+    return (root, query, cb) ->
+        cb.and(
+            cb.equal(root.get(SpecField.ORGANIZATION), _os.getCurrentOrganization()),
+            cb.or(
+                cb.isNull(root.get(SpecField.FACILITY)), // null check first
```

Comment:
I would remove the comment, since actually postgresql doesn't care which order you do those things. 😁

Reply:
It actually matters for the following line:
`cb.equal(root.get(SpecField.FACILITY).get(SpecField.INTERNAL_ID), facilityId)));`
That `.get()` will fail.

Reply:
But I made the comment a bit more clear.